python/pyspark/sql/connect/functions/builtin.py

```diff
@@ -2578,6 +2590,16 @@ def regexp_like(str: "ColumnOrName", regexp: "ColumnOrName") -> Column:
 regexp_like.__doc__ = pysparkfuncs.regexp_like.__doc__
 
 
+def randstr(length: "ColumnOrName", seed: Optional["ColumnOrName"] = None) -> Column:
+    if seed is None:
+        return _invoke_function_over_columns("randstr", length)
```

Comment:
```suggestion
        return _invoke_function_over_columns("randstr", length, lit(random.randint(0, sys.maxsize)))
```

We need to explicitly set the `seed` in connect clients, to avoid making the output dataframe non-deterministic (see https://github.com/apache/spark/commit/14ba4fc479155611ca39bda6f879c34cc78af2ee)

Reply:
👍  thanks for pointing this out!